examples/new_connection_pool.go

```diff
@@ -0,0 +1,49 @@
+//go:build new_connection_pool
+// +build new_connection_pool
+
+package main
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/chaitin/t1k-go"
+)
+
+func initDetect(addr string) *t1k.ChannelPool {
```

Comment:
在 `examples/new_connection_pool.go` 中，`initDetect` 函数在初始化连接池失败时返回 `nil`。建议返回具体的错误信息，以便更好地调试。

```suggestion
func initDetect(addr string) (*t1k.ChannelPool, error) {
	pc := &t1k.PoolConfig{
		InitialCap:  1,
		MaxIdle:     16,
		MaxCap:      32,
		Factory:     &t1k.TcpFactory{Addr: addr},
		IdleTimeout: 30 * time.Second,
	}
	server, err := t1k.NewChannelPool(pc)
	if err != nil {
		return nil, err
	}
	return server, nil
}